xrpl/transaction/types/mptoken_metadata.go

```diff
@@ -1,5 +1,106 @@
 package types
 
-func MPTokenMetadata(value string) *string {
-	return &value
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+const (
+	// MPTokenMetadataMaxSize defines the maximum allowed size in bytes for
+	// multi-purpose token metadata when serialized to JSON.
+	// https://github.com/XRPLF/XRPL-Standards/tree/master/XLS-0089d-multi-purpose-token-metadata-schema
+	MPTokenMetadataMaxSize = 1024
+)
+
+// MPTokenMetadataURL represents a URL reference within multi-purpose token metadata.
+// It follows the XLS-0089d standard for multi-purpose token metadata schema.
+type MPTokenMetadataURL struct {
+	URL   string `json:"url"`   // The URL address
+	Type  string `json:"type"`  // The MIME type of the resource
+	Title string `json:"title"` // Human-readable title for the URL
+}
+
+// MPTokenMetadata represents the metadata structure for multi-purpose tokens
+// following the XLS-0089d standard. This structure defines the schema for
+// token metadata that can be attached to XRPL tokens.
+type MPTokenMetadata struct {
+	Ticker         string               `json:"ticker"`                    // Short symbol for the token
+	Name           string               `json:"name"`                      // Full name of the token
+	Desc           string               `json:"desc,omitempty"`            // Description of the token
+	Icon           string               `json:"icon,omitempty"`            // URL to token icon image
+	AssetClass     string               `json:"asset_class"`               // Primary classification of the asset
+	AssetSubclass  string               `json:"asset_subclass"`            // Secondary classification of the asset
+	IssuerName     string               `json:"issuer_name"`               // Name of the token issuer
+	URLs           []MPTokenMetadataURL `json:"urls,omitempty"`            // Additional URL references
+	AdditionalInfo json.RawMessage      `json:"additional_info,omitempty"` // Custom additional metadata
+}
+
+// Validate checks if the MPTokenMetadata has all required fields populated.
+// Required fields are: Ticker, AssetClass, AssetSubclass, and Name.
+// Returns an error if any required field is empty or contains only whitespace.
+func (m MPTokenMetadata) Validate() error {
```

Comment:
There are non-allowed values that are not checked. See https://github.com/XRPLF/XRPL-Standards/tree/master/XLS-0089-multi-purpose-token-metadata-schema#2-base-metadata-schema

Reply:
`Allowed Values` column contains all details about how to validate each `MPTokenMetadata` values